client/components/disputed-order-notice/index.js

```diff
@@ -0,0 +1,225 @@
+import moment from 'moment';
+import React, { useEffect } from 'react';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { dateI18n } from '@wordpress/date';
+import { createInterpolateElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import InlineNotice from 'components/inline-notice';
+import { formatExplicitCurrency } from 'utils/currency';
+import { reasons } from 'wcpay/disputes/strings';
+import { getDetailsURL } from 'wcpay/components/details-link';
+import {
+	isAwaitingResponse,
+	isInquiry,
+	isUnderReview,
+} from 'wcpay/disputes/utils';
+import { useCharge } from 'wcpay/data';
+import wcpayTracks from 'tracks';
+import './style.scss';
+
+const DisputedOrderNoticeHandler = ( { chargeId, onDisableOrderRefund } ) => {
+	const { data: charge } = useCharge( chargeId );
+	const disputeDetailsUrl = getDetailsURL( chargeId, 'transactions' );
+
+	// Disable the refund button if there's an active dispute.
+	useEffect( () => {
+		const { dispute } = charge;
+		if ( ! charge?.dispute ) {
+			return;
+		}
+		// Refunds are only allowed if the dispute is an inquiry or if it's won.
+		const isRefundable =
+			isInquiry( dispute ) || [ 'won' ].includes( dispute.status );
+		if ( ! isRefundable ) {
+			onDisableOrderRefund( dispute.status );
+		}
+	}, [ charge, onDisableOrderRefund ] );
+
+	const { dispute } = charge;
+	if ( ! charge?.dispute ) {
+		return null;
+	}
+
+	// Special case the "under review" notice which is much simpler.
+	// (And return early.)
+	if ( isUnderReview( dispute.status ) ) {
+		<DisputeOrderLockedNotice
+			message={ __(
+				'This order has an active payment dispute. Refunds and order editing are disabled.',
```

Comment:
Note I tweaked this wording to remove redundant "at this time". 

<img width="1096" alt="Screenshot 2023-10-30 at 2 33 24 PM" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/37de4408-0573-4d15-aab2-5de08bf4b8ed">

Reply:
Was: `This order has an active payment dispute. Refunds and order editing are disabled at this time.`